pkg/actuators/machine/reconciler_test.go

```diff
@@ -171,121 +170,266 @@ func TestSetMachineCloudProviderSpecifics(t *testing.T) {
-		if err != nil {
-			t.Fatal(err)
-		}
-		encodedProviderConfig, err := RawExtensionFromProviderSpec(tc.providerConfig)
-		if err != nil {
-			t.Fatalf("Unexpected error")
-		}
-		providerStatus, err := RawExtensionFromProviderStatus(stubProviderStatus(powerVSProviderID))
-		if err != nil {
-			t.Fatalf("Failed to set providerStatus")
-		}
-		machine.Spec.ProviderSpec = machinev1beta1.ProviderSpec{Value: encodedProviderConfig}
-		machine.Status.ProviderStatus = providerStatus
+			providerConfig, err := RawExtensionFromProviderSpec(tc.providerConfig)
+			if err != nil {
+				t.Fatalf("Unexpected error")
+			}
 
-		fakeClient := fake.NewFakeClientWithScheme(scheme.Scheme, machine, tc.powerVSCredentialsSecret, tc.userDataSecret)
+			providerStatus, err := RawExtensionFromProviderStatus(&tc.providerStatus)
+			if err != nil {
+				t.Fatal(err)
+			}
 
-		machineScope, err := newMachineScope(machineScopeParams{
-			client:  fakeClient,
-			machine: machine,
-			powerVSClientBuilder: func(client runtimeclient.Client, secretName, namespace, cloudInstanceID string,
-				debug bool) (client.Client, error) {
-				return mockPowerVSClient, nil
-			},
-			powerVSMinimalClient: func(client runtimeclient.Client) (client.Client, error) {
-				return nil, nil
-			},
-		})
-		if err != nil {
-			t.Fatal(err)
-		}
+			machine := tc.machineFunc()
+			machine.Spec.ProviderSpec = machinev1beta1.ProviderSpec{Value: providerConfig}
+			machine.Status.ProviderStatus = providerStatus
 
-		reconciler := newReconciler(machineScope)
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(machine, powerVSCredentialsSecret, userDataSecret).Build()
+			mockPowerVSClient := tc.powerVSClientFunc(ctrl)
 
-		// test create
-		err = reconciler.create()
-		log.Printf("Error is %v", err)
-		if tc.expectedError != nil {
-			if err == nil {
-				t.Error("reconciler was expected to return error")
-			}
-			if err != nil && err.Error() != tc.expectedError.Error() {
-				t.Errorf("Expected: %v, got %v", tc.expectedError, err)
-			}
-		} else {
+			machineScope, err := newMachineScope(machineScopeParams{
+				client:  fakeClient,
+				machine: machine,
+				powerVSClientBuilder: func(client runtimeclient.Client, secretName, namespace, cloudInstanceID string,
+					debug bool) (client.Client, error) {
+					return mockPowerVSClient, nil
+				},
+				powerVSMinimalClient: func(client runtimeclient.Client) (client.Client, error) {
+					return nil, nil
+				},
+			})
 			if err != nil {
-				t.Errorf("reconciler was not expected to return error: %v", err)
+				t.Fatal(err)
 			}
-		}
+			reconciler := newReconciler(machineScope)
+
+			err = reconciler.create()
+			if tc.expectError {
+				if err == nil {
+					t.Fatal("create machine expected to return an error")
+				}
+				if reconciler.providerStatus != nil && reconciler.providerStatus.InstanceID != nil {
+					if *reconciler.providerStatus.InstanceID != *stubInstanceWithBuildState().PvmInstanceID {
```

Comment:
It better we feed this `*stubInstanceWithBuildState().PvmInstanceID` data in the testCases instead of reading directly in the testcase loop